### PR TITLE
Add file object argument to adimage

### PR DESCRIPTION
**Pull request by tomako**
Fix for issue #58

### CODE REVIEW

**Comment by tomako** [comment]
@ebzlo any feedback?

File: facebookads/objects.py [hunk 1]
```diff
@@ -1357,15 +1358,24 @@ def get_endpoint(cls):
         return 'adimages'
 
     @classmethod
-    def remote_create_from_zip(cls, filename, parent_id, api=None):
+    def remote_create_from_zip(cls, file=None, parent_id=None, api=None,
+                               filename=None):
+        if filename:
+            warnings.warn("The 'filename' argument has been depricated in "
```
Comment (ebzlo) [inline facebookads/objects.py]: nit: deprecated

File: facebookads/objects.py [hunk 2]
```diff
@@ -1357,15 +1358,24 @@ def get_endpoint(cls):
...
+                          "favor of 'file'", DeprecationWarning, stacklevel=2)
+        file = file or filename
```
Comment (ebzlo) [inline facebookads/objects.py]: This may cause unexpected behavior for someone who is not diligent with understanding the different parameters. Instead, let's throw an error if the both `file` or `filename` are provided. In the event someone does it (where they represent different files), we silently select one for them where we should probably just complain.

```
if filename:
    # warn
    file = filename
    if file:
        throw

if isinstance(file, basestring):
    ...
```

File: facebookads/objects.py [hunk 3]
```diff
@@ -1440,6 +1450,11 @@ def _set_data(self, data):
         else:
             return super(AdImage, self)._set_data(data)
 
+    def export_data(self):
+        """Since AdImage is not updatable there's no need to collect
+        changed properties"""
+        return {}
```
Comment (ebzlo) [inline facebookads/objects.py]: I'm not sure I agree with change. Is this required for this diff? If not, maybe submit it in a different PR to have a conversation around it. :)

File: facebookads/objects.py [hunk 4]
```diff
@@ -1452,23 +1467,31 @@ def remote_create(
...
-                files={filename: open_file},
-                batch=batch,
-                failure=failure,
-                params=params,
-                success=success,
-                api_version=api_version,
+                "AdImage required a file or filename to be defined."
             )
+        if self.get(self.Field.filename):
+            warnings.warn("The 'filename' property has been depricated in "
```
Comment (ebzlo) [inline facebookads/objects.py]: nit: deprecated

File: facebookads/objects.py [hunk 5]
```diff
@@ -1452,23 +1467,31 @@ def remote_create(
...
+                          "favor of 'file'", DeprecationWarning, stacklevel=2)
+        file = self.get(self.Field.file, self.get(self.Field.filename))
```
Comment (ebzlo) [inline facebookads/objects.py]: `self.get(self.Field.filename)` would never return a `file` instance, right?

**Comment by ebzlo** [comment]
Thanks for the PR! Sorry for the delay. This is great! I have left a couple comments.

**Comment by caro5** [comment]
Hi! We are using the Ads SDK and it would be awesome if this was merged soon. Is there a timeline for this?
